src/components/boardComponents/useItemActions.jsx

```diff
@@ -89,70 +90,120 @@ export const useItemActions = () => {
+      ...selectedItemList.map(
+        ({ x, id }) => x + document.getElementById(id).clientWidth
+      )
+    );
+    minMax.max.y = Math.max(
+      ...selectedItemList.map(
+        ({ y, id }) => y + document.getElementById(id).clientHeight
+      )
+    );
+    const { clientWidth, clientHeight } = document.getElementById(selectedItemList[0].id);
+    let newX = minMax.min.x + (minMax.max.x - minMax.min.x) / 2 - clientWidth / 2;
+    let newY = minMax.min.y + (minMax.max.y - minMax.min.y) / 2 - clientHeight / 2; 
+    
 
-      const [newX, newY] = [
-        (minMax.min.x + minMax.max.x) / 2,
-        (minMax.min.y + minMax.max.y) / 2,
-      ];
-      let index = -1;
       batchUpdateItems(selectedItems, (item) => {
-        index += 1;
-        const { clientWidth, clientHeight } = document.getElementById(item.id);
-        return {
+        const newItem = {
           ...item,
-          x: newX - clientWidth / 2 + index,
-          y: newY - clientHeight / 2 - index,
+          x: newX,
+          y: newY,
         };
+        newX += stackThickness;
+        newY -= stackThickness;
+        return newItem;
       });
-    },
-    [getSelectedItemList, batchUpdateItems, selectedItems]
-  );
+  },
+  [getSelectedItemList, batchUpdateItems, selectedItems]
+);
+
+// Stack selection to Top Left
+const stackToTopLeft = useRecoilCallback(
+  ({ snapshot }) => async () => {
+    const selectedItemList = await getSelectedItemList(snapshot);
+
+    let newX = selectedItemList[0].x;
+    let newY = selectedItemList[0].y;
+    let stackThickness = 0;
+
+    if (selectedItemList.length >= 32) {
+        stackThickness = 0.5;
+      } else {
+        stackThickness = 1;
+      }
+    
+    batchUpdateItems(selectedItems, (item) => {
+      const newItem = {
+        ...item,
+        x: newX,
+        y: newY,
+      };
+      newX += stackThickness;
+      newY -= stackThickness;
+      return newItem;
+    });
+  },
+  [getSelectedItemList, batchUpdateItems, selectedItems]
+);
 
   // Align selection to a line
   const alignAsLine = useRecoilCallback(
     ({ snapshot }) => async () => {
       const selectedItemList = await getSelectedItemList(snapshot);
 
-      // Compute
-      const minMax = { min: {}, max: {} };
-      minMax.min.x = Math.min(...selectedItemList.map(({ x }) => x));
-      minMax.min.y = Math.min(...selectedItemList.map(({ y }) => y));
-      minMax.max.x = Math.max(
-        ...selectedItemList.map(
-          ({ x, id }) => x + document.getElementById(id).clientWidth
-        )
-      );
-      minMax.max.y = Math.max(...selectedItemList.map(({ y }) => y));
+      // Count number of elements
+      const numberOfElements = selectedItemList.length;
 
-      const [newX, newY] = [minMax.min.x, minMax.max.y];
-      let index = -1;
+      let newX = selectedItemList[0].x;
+      let newY = selectedItemList[0].y;
+      let index = numberOfElements;
+      let gapBetweenItems = 5; // Negative value is possible.
```

Comment:
Can you make it a function parameter with a default value? I plan to add parametrized actions (to avoid duplicate rotation action for each angle for example).